bitnami/nginx/templates/deployment.yaml

```
@@ -252,21 +252,21 @@ spec:
           {{- else if .Values.livenessProbe.enabled }}
           livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.livenessProbe "enabled") "context" $) | nindent 12 }}
             tcpSocket:
-              port: {{ ternary "http" "https" .Values.tls.enabled }}
+              port: {{ ternary "http" "https" (empty .Values.containerPorts.https) }}
```

Your proposal could be enough but keeping in mind how we are setting the ports some lines above and the proposal done in #20934, I'd use something like (not tested):
```suggestion
              port: {{ ternary "https" "http" (and (not  .Values.containerPorts.http)  .Values.containerPorts.https) }}
```

WDYT?

---

Makes sense to me, however I get nothing but `wrong type for value; expected bool; got string` if I don't use `empty`, so I've pushed a commit that combines both.  Tested it and the `https` port is used _if and only if_ `containerPorts.http` is empty and `containerPorts.https` is set, else all combinations result in `http` being used.